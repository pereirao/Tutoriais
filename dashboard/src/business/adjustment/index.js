// @flow

export type Adjustment = {
  id: number,
  created_at: string, // rails timestamp
  amount: number,
  credit: boolean,
  reason: string,
  description: string
};

export * as adjustment_actions from './actions';
export * as adjustment_epics from './epics';
export * as adjustment_helpers from './helpers';
export { default as adjustmentReducer } from './reducer';
export { default as adjustment_selectors } from './selectors';
