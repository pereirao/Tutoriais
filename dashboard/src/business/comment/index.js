// @flow
export type Author = {
  email: string
};

export type Comment = {
  id: number,
  created_at: string, // rails timestamp
  note: string,
  author: Author,
  posted_as: string
};

export * as comment_actions from './actions';
export * as comment_epics from './epics';
export * as comment_helpers from './helpers';
export { default as commentReducer } from './reducer';
export { default as comment_selectors } from './selectors';
