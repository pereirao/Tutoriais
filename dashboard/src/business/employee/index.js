// @flow

type EmployeeRole = 'supplier' | 'driver';

export type Employee = {
  id: number,
  first_name: string,
  last_name: string,
  active: boolean,
  email: string,
  roles: Array<EmployeeRole>
};

export * as employee_epics from './epics';
export * as employee_actions from './actions';
export * as employee_helpers from './helpers';
export { default as employeeReducer } from './reducer';
export { default as employee_selectors } from './selectors';
