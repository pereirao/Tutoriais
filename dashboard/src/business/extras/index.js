export * from './extras.dux';
export * from './extras.sagas';
