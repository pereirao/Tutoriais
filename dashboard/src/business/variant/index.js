// @flow

export type Variant = {
  id: number,
  name: string,
  volume: string,
  inventory: number,
  sku: string,
  active: boolean,
  price: number,
  sale_price: number,
  category: string
};

export * as variant_epics from './epics';
export * as variant_actions from './actions';
export * as variant_helpers from './helpers';
export { default as variantReducer } from './reducer';
export { default as variant_selectors } from './selectors';

