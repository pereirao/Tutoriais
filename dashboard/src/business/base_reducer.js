//@flow

import {combineReducers} from 'redux';
import {adjustmentReducer} from './adjustment';
import {commentReducer} from './comment';
import {employeeReducer} from './employee';
import {notificationReducer} from './notification';
import {notificationMethodReducer} from './notification_method';
import {orderReducer} from './order';
import {reportReducer} from './report';
import {uiReducer} from './ui';
import {sessionReducer} from './session';
import {variantReducer} from './variant';
import {settingsReducer} from './settings';
import substituteReducer from './substitute/substitute.dux';
import productReducer from './product/product.dux';
import {connectRouter} from "connected-react-router";

const appReducer = (history) => combineReducers({
  adjustment: adjustmentReducer,
  comment: commentReducer,
  employee: employeeReducer,
  notification: notificationReducer,
  notification_method: notificationMethodReducer,
  order: orderReducer,
  report: reportReducer,
  ui: uiReducer,
  router: connectRouter(history),
  session: sessionReducer,
  variant: variantReducer,
  settings: settingsReducer,
  substitute: substituteReducer,
  product: productReducer
});

const rootReducer = (history) => (state, action) => {
  let app_state = state;
  if (action.type === 'SESSION:RESET_STATE') {
    app_state = undefined;
  }
  return appReducer(history)(app_state, action);
};

export default rootReducer;
