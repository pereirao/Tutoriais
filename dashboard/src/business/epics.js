import _ from 'lodash';

import { combineEpics } from 'redux-observable';

import { adjustment_epics } from './adjustment';
import { comment_epics } from './comment';
import { order_epics } from './order';
import { session_epics } from './session';
import { employee_epics } from './employee';
import { notification_method_epics } from './notification_method';
import { report_epics } from './report';
import { variant_epics } from './variant';
import { settings_epics } from './settings';

// convert the imported epic objects into an array
const epics_by_module = [
  adjustment_epics,
  comment_epics,
  order_epics,
  session_epics,
  employee_epics,
  notification_method_epics,
  report_epics,
  variant_epics,
  settings_epics
];

const epics = _.flatMap(epics_by_module, (module_epic_exports) => {
  // we pull the actual epic functions out of the exports we recieved from the each file
  const module_epics = _.values(module_epic_exports);
  return module_epics;
});

const rootEpic = combineEpics(...epics);
export default rootEpic;
