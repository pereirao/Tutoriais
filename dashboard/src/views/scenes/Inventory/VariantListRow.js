// @flow

import React, {Component} from 'react';
import {connect} from 'react-redux';
import EditVariantForm from './EditVariantForm';
import style_sheet from './Inventory.module.css';
import MBTable from '../../elements/MBTable';
import MBButton from '../../elements/MBButton';
import MBIcon from '../../elements/MBIcon';
import I18n from '../../../localization';
import type {Variant} from '../../../business/variant';
import {variant_helpers, variant_actions, variant_selectors} from '../../../business/variant';
import MergeProductModal from './MergeProductModal';

const {variantTableRow, VARIANTS_COLUMN_ORDER} = variant_helpers;

type VariantListRowProps = { variant: Variant, updating: boolean };
type VariantListRowState = { edit_mode: boolean };

export class VariantListRow extends Component {
  props: VariantListRowProps
  state: VariantListRowState

  constructor(props: VariantListRowProps) {
    super(props);
    this.state = {edit_mode: false};
  }

  toggleEditMode = () => {
    this.setState({edit_mode: !this.state.edit_mode});
  };

  clickAddButton = () => {
    const {updateVariant, variant} = this.props;
    updateVariant({variant_id: variant.id, sku: variant.sku, not_owned: false});
  }

  clickRemoveButton = () => {
    const {updateVariant, variant} = this.props;
    updateVariant({variant_id: variant.id, sku: variant.sku, active: false});
  }

  clickRestoreButton = () => {
    const {updateVariant, variant} = this.props;
    updateVariant({variant_id: variant.id, sku: variant.sku, active: true});
  }

  onChange = (e: Object, field: string) => {
    this.setState({[field]: e.target.value});
  };

  activateButton = (isMobile) => {
    if (this.props.variant.not_owned) {
      return (
        <MBButton onClick={this.clickAddButton} style={{backgroundColor: '#F6FAF3', color: '#84B168', border: 'none'}}>
          <div className={style_sheet.editIcon}>
            <MBIcon icon="add" color="mb_green" size="small"/>
          </div>
          {isMobile ? 'Add' : 'Add to Inventory'}
        </MBButton>
      );
    } else if (this.props.variant.active) {
      return (
        <MBButton onClick={this.clickRemoveButton}
                  style={{backgroundColor: '#FAF6F4', color: '#AE7B7B', border: 'none'}}>
          <div className={style_sheet.editIcon}>
            <MBIcon icon="delete" color="mb_red" size="small"/>
          </div>
          {isMobile ? 'Remove' : 'Remove from Inventory'}
        </MBButton>
      );
    } else if (this.props.variant.product_state === 'pending' || this.props.variant.product_state === 'inactive') {
      return (
        <MergeProductModal variant={this.props.variant}/>
      );
    } else {
      return (
        <MBButton onClick={this.clickRestoreButton} button_type="link">
          <div className={style_sheet.editIcon}>
            <MBIcon icon="cancel" color="mb_red" size="small"/>
          </div>
          Restore
        </MBButton>
      );
    }
  }

  renderCells(col: Object) {
    const row_data = variantTableRow(this.props.variant);
    if (this.props.updating === this.props.variant.sku) return 'loading...'; // TODO: Use Loader
    switch (col) {
      case 'product_name': {
        return (
          <span>
            <span>
              {row_data.product_name}
            </span>
            <br/>
            <span className={style_sheet.productDetails}>
              {row_data.product_details}
            </span>
          </span>
        );
      }
      case 'edit': {
        return (this.renderButton());
      }
      case 'mobile_summary': {
        return (
          <span className={style_sheet.mobileSummaryWrapper}>
            <div>
              <span>
                {I18n.t(`ui.table.variant_mobile_main_on_sale.${row_data.sale_price !== 'Not discounted'}`, {
                  name: row_data.product_name,
                  sale_price: row_data.sale_price,
                  price: row_data.price
                })}
              </span>
              <br/>
              <span className={style_sheet.productDetails}>
                {I18n.t('ui.table.variant_mobile_secondary', {
                  details: row_data.product_details,
                  inventory: row_data.inventory
                })}
              </span>
            </div>
            <div className={style_sheet.mobileEditButtonWrapper}>
              <span>
                {this.renderButton(true)}
              </span>
              <span>
                {this.activateButton(true)}
              </span>
            </div>
          </span>
        );
      }
      case 'active': {
        return this.activateButton();
      }
      default:
        return (row_data[col]);
    }
  }

  renderButton(isMobile) {
    if (this.props.updating === this.props.variant.sku) {
      return (<td>Loading...</td>); // TODO: Use Loader
    } else {
      return (
        <span className={style_sheet.editButtonWrapper}>
          <MBButton button_type="link" onClick={this.toggleEditMode}>
            <div className={style_sheet.editIcon}>
              <MBIcon icon="edit" color="mb_red" size="small"/>
            </div>
            {isMobile ? I18n.t('ui.button.mobile_edit_product') : I18n.t('ui.button.edit_product')}
          </MBButton>
        </span>
      );
    }
  }

  render() {
    if (this.state.edit_mode) {
      return (
        <EditVariantForm toggleEditMode={this.toggleEditMode} variant={this.props.variant}/>
      );
    } else {
      return (
        <MBTable.Row key={`${this.props.variant.sku}row`}>
          {VARIANTS_COLUMN_ORDER.map(col => {
            return (
              <MBTable.Cell
                key={`${this.props.variant.sku}${col}`}
                mobile_only={col === 'mobile_summary'}
                mobile_hidden={col !== 'mobile_summary'}>
                {this.renderCells(col)}
              </MBTable.Cell>
            );
          })}
        </MBTable.Row>
      );
    }
  }
}

const VariantListRowSTP = (state) => ({
  updating: variant_selectors.isUpdating(state)
});
const VariantListRowDTP = {
  updateVariant: variant_actions.updateVariant
};

const VariantListRowContainer = connect(VariantListRowSTP, VariantListRowDTP)(VariantListRow);

export default VariantListRowContainer;
