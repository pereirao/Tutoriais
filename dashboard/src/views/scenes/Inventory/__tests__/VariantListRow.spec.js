import React from 'react';

import { VariantListRow } from '../VariantListRow';
import { variant_helpers } from '../../../../business/variant';
import variant_factory from '../../../../business/variant/__tests__/variant.factory';


const { VARIANTS_COLUMN_ORDER } = variant_helpers;
const VARIANT = variant_factory.build({id: 1});

it('renders correctly', () => {
  expect(shallow(<VariantListRow key={VARIANT.id} columns={VARIANTS_COLUMN_ORDER} variant={VARIANT} />)).toMatchSnapshot();
});
