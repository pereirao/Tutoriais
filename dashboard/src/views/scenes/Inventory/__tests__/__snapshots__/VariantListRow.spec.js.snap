// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Row>
  <Cell
    mobile_hidden={false}
    mobile_only={true}
  >
    <span
      className="mobileSummaryWrapper"
    >
      <div>
        <span>
          [missing {{name}} value] - 19.99
        </span>
        <br />
        <span
          className="productDetails"
        >
          undefined | undefined - 10 in stock
        </span>
      </div>
      <div
        className="mobileEditButtonWrapper"
      >
        <span>
          <span
            className="editButtonWrapper"
          >
            <MBButton
              button_type="link"
              onClick={[Function]}
            >
              <div
                className="editIcon"
              >
                <MBIcon
                  color="mb_red"
                  icon="edit"
                  size="small"
                />
              </div>
              Edit Product
            </MBButton>
          </span>
        </span>
      </div>
    </span>
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    this_is_a_sku
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    <span>
      <span />
      <br />
      <span
        className="productDetails"
      >
        undefined | undefined
      </span>
    </span>
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    No
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    10
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    19.99
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    Not discounted
  </Cell>
  <Cell
    mobile_hidden={true}
    mobile_only={false}
  >
    <span
      className="editButtonWrapper"
    >
      <MBButton
        button_type="link"
        onClick={[Function]}
      >
        <div
          className="editIcon"
        >
          <MBIcon
            color="mb_red"
            icon="edit"
            size="small"
          />
        </div>
        Edit Product
      </MBButton>
    </span>
  </Cell>
</Row>
`;
